README.md

```diff
@@ -525,6 +525,38 @@ or with a light image:
 docker run -v /path/to/models:/models ghcr.io/ggerganov/llama.cpp:light -m /models/7B/ggml-model-q4_0.bin -p "Building a website can be done in 10 simple steps:" -n 512
 ```
 
+### Docker With CUDA
+
+Assuming one has the [nvidia-container-toolkit](https://github.com/NVIDIA/nvidia-container-toolkit) properly installed on Linux, or is using a GPU enabled cloud, `cuBLAS` should be accessible inside the container.
+
+#### Building Locally
+
+```bash
+docker build -t local/llama.cpp:full-cuda -f .devops/full-cuda.Dockerfile .
+docker build -t local/llama.cpp:light-cuda -f .devops/main-cuda.Dockerfile .
+```
+
+You may want to pass in some different `ARGS`, depending on the CUDA environment supported by your container host, as well as the GPU architecture.
+
+The defaults are:
+
+- `CUDA_VERSION` set to `11.7.1`
+- `CUDA_DOCKER_ARCH` set to `all`
+
+The resulting images, are essentially the same as the non-CUDA images:
+
+1. `local/llama.cpp:full-cuda`: This image includes both the main executable file and the tools to convert LLaMA models into ggml and convert into 4-bit quantization.
+2. `local/llama.cpp:light-cuda`: This image only includes the main executable file.
+
+#### Usage
+
+After building locally, Usage is similar to the non-CUDA examples, but you'll need to add the `--gpus` flag. You will also want to use the `--n-gpu-layers` flag.
+
+```bash
+docker run --gpus all -v /path/to/models:/models local/llama.cpp:full --run -m /models/7B/ggml-model-q4_0.bin -p "Building a website can be done in 10 simple steps:" -n 512 --n-gpu-layers 1
+docker run --gpus all -v /path/to/models:/models local/llama.cpp:light -m /models/7B/ggml-model-q4_0.bin -p "Building a website can be done in 10 simple steps:" -n 512 --n-gpu-layers 1
```

Comment:
You forgot to rename the containers here.
Also I get the following error when I try to run the containers:
```
/home/johannesg/Projects/llama.cpp [git::feat/docker-cuda *] [johannesg@johannes-pc] [17:45]
> docker run --gpus all -v /mnt/Windows10/Users/Johannes/Projects:/models local/llama.cpp:full-cuda --run -m /models/llama-7b-ggml-q4_0.bin -p "Building a website can be done in 10 simple steps:" -n 512 --n-gpu-layers 99
error: unknown argument: a
usage: ./main [options]
```

Reply:
```
/home/johannesg/Projects/llama.cpp [git::feat/docker-cuda *] [johannesg@johannes-pc] [18:04]
> docker run --gpus all -v /mnt/Windows10/Users/Johannes/Projects:/models local/llama.cpp:light-cuda --run -m /models/llama-7b-ggml-q4_0.bin -p "Building a website can be done in 10 simple steps:" -n 512 --n-gpu-layers 99
error: unknown argument: --run
usage: /main [options]
```

Reply:
@JohannesGaessler - Good catch on the label rename, fixed.

With regards to the second issue, I've left this [comment here](https://github.com/ggerganov/llama.cpp/pull/1461#issuecomment-1547050032), and there's also an [open issue here](https://github.com/ggerganov/llama.cpp/issues/1463) on the subject.

I mirrored what was already in the `README.md` so that issue can be resolved separately, as the problem exists in the non-CUDA instructions as well. I aimed for alignment with the other documentation, but validated with `--random-prompt` instead of `-p "something something"`.

I hope a fix for the Docker documentation can be found, that preserves the brevity of the Docker examples and doesn't turn into a treatise on bash & docker string escaping :) ... It's something else I want to look into fixing when I have time, but separately from this functionality.

Reply:
@JohannesGaessler - In your second failed docker command:

It appears you're attempting to run the `full` arguments on the `light` container.

```
> docker run --gpus all -v /mnt/Windows10/Users/Johannes/Projects:/models local/llama.cpp:light-cuda --run -m /models/llama-7b-ggml-q4_0.bin -p "Building a website can be done in 10 simple steps:" -n 512 --n-gpu-layers 99
error: unknown argument: --run
```

The `--run` argument is not available in either the `light` or `light-cuda` examples. This behavior matches the original docker image as well.

Reply:
Thank you for the hint regarding the second command. It now works. What I've noticed is that CTRL+C cannot be used to interrupt the token generation (but I assume that is then an issue on master as well?). In any case, I got both containers to work with GPU acceleration and there don't seem to be issues that don't exist on master then.

Reply:
I run into that issue with ctrl+c in Docker as well, regardless of if it's the CUDA or non-CUDA version.

It does however work, when I'm in interactive mode and I press it twice. I don't know what the root cause of it is yet, but when I get a chance, I plan to look into it, because killing the container manually is a hassle.

Reply:
Intuitively I would have thought that starting a shell inside the docker container and then executing `./main` as you would normally would fix CTRL+C behavior. CLI arg escaping probably too. Would only be good for CLI use though.